### PR TITLE
Role Resource admin created

### PR DESCRIPTION
**Pull request by zwejra**
New role RESOURCEADMIN added.
Methods for adding/removing admin to/from resource added to AuthzResolverImpl and to ResourcesManager.
Method isResourceAdmin added to AuthzResolver API.
Method getAdminGroups added to ResourcesManager.
Privileges to assigning/removing groups to/from resource added to role Resource admin.
AlreadyAdminException now supports creating exception containing resource.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java [hunk 1]
```diff
@@ -453,6 +458,9 @@ public static boolean isAuthorizedForAttribute(PerunSession sess, ActionType act
 			if(roles.contains(Role.FACILITYADMIN)) {
 				if(isAuthorized(sess, Role.FACILITYADMIN, resource)) return true;
 			}
+			if(roles.contains(Role.RESOURCEADMIN)) {
```
Comment (zwejra) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java]: Should it be here in isAuthorizedForAttribute method?

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java [hunk 2]
```diff
@@ -290,7 +265,8 @@ public void assignGroupToResource(PerunSession sess, Group group, Resource resou
 		getResourcesManagerBl().checkResourceExists(sess, resource);
 
 		// Authorization
-		if (!AuthzResolver.isAuthorized(sess, Role.VOADMIN, resource)) {
+		if (!AuthzResolver.isAuthorized(sess, Role.VOADMIN, resource) &&
+				!AuthzResolver.isAuthorized(sess, Role.RESOURCEADMIN, resource)) {
```
Comment (zwejra) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntry.java]: I suppose VOADMIN is still authorized for actions he had before.

**Comment by licehammer** [comment]
Hi @zwejra, 

to answer your aforementioned questions:
1. copyAttributes method is used only to copy Attribute objects. Other setting and relations remains unchanged.  Therefore you shouldn't copy admins.
2. VO manager should be able to list and change resources managers for all resources within his VO. Resource manger can do the same for his resource. Other roles can't see or manipulate resource managers.

**Comment by stavamichal** [comment]
There is missing RPC part of methods "getAdmin, getAdminGroups" etc.

**Comment by zwejra** [comment]
Hi @licehammer and @stavamichal,

I think its done. Getters were created, RPC methods as well. Tests were created too.
Please check it, if its all okay.

**Pull request merged**